LiveSched/src/main/java/dev/coms4156/project/livesched/Task.java

```diff
@@ -0,0 +1,117 @@
+package dev.coms4156.project.livesched;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+/**
+ * Represents a task that has to be done. 
+ * This class stores the id, resources needed, and the location of the task.
+ */
+public class Task {
+  private final String taskId;
+  private Map<ResourceType, Integer> resourceList; // Key = ResoureceType, Value = Units needed
+  private int priority; 
+  private LocalDateTime startTime;
+  private LocalDateTime endTime;
+  private double latitude;
+  private double longitude;
+
+  /**
+   * Constructs a new Task object with the given parameters.
+   *
+   * @param taskId        the unique ID of the resource
+   * @param resourceList  the map of resources and their quantity needed
+   * @param priority      the priority of the task
+   * @param endTime       the time that task ends
+   * @param latitude      the latitude of the resource's location
```

Comment:
task?

Reply:
Does the task require a location at all? I don't think its needed

Reply:
Hmm, that's a great question. I think in ER cases, we might want to get the patient's current location (which could be the task's location?) so that we can assign them to the right hospital that is close and has the available resources. I guess, in general, if we want to include proximity in the scheduling algorithm, we would somehow need the location information for both the task and resource.

Reply:
Yes, I thought we would need location for both task and resources, otherwise it wouldn't really make sense to have it for resources. Maybe we would need to check if resources needed are available 'in the right location' (compare the location of task and resources) instead of just checking the available numbers?

Reply:
I think what I was picturing in my head was a service that allows the client to find a ride to get to the destination (Resource). But I feel like we're also generalizing the "task" we want to do (either ride or just finding out the availability). Should we clarify this?

Reply:
Maybe we should align on the goal of the service. What is the Task we are trying to schedule with the Resource?

Reply:
I might be going off tangent with this but if we are still going with the idea of using Uber/Lyft API, I think we should keep this in mind when creating the classes. For instance, where are the "source" and "destination" located? And based on the availability/resource of the destination, the "scheduler" will link up the appropriate destination with the source and call the Uber/Lyft API. Maybe the naming for Task and Resource is confusing us.

Reply:
Then my next question is: how are we going to define the availability of the resource? I think we should be calling an API, something like a Google Maps API if possible to get information about the destination, or we could just use a dummy data and pretend everything exists in our database.

Reply:
My thought was that a resource could be anything from a car/driver to a bed or doctor. Task represents the job that requires one or more resources. It could be transferring a patient, taking care of a patient, scheduling surgery, etc. (examples in a hospital setting). The availability of a resource can be simply defined as whether it is assigned to a task or not (free). You can check it with isAvailableAt (in the Resource class) using the task's start time as findAvailableResource (in the ResourceType class) does. I think that should be done first, and then we can use some algorithm using proximity, priority, etc. to assign a specific resource from the found available resources to a task. For location information, I just used latitude, longitude for simplicity, but we can definitely use another API.

Reply:
By information, I meant availability not location but yeah isAvailableAt makes perfect sense. Sorry I missed that! I think I'm just confused bc of what TA said

Reply:
That's okay! I'm also confused here haha. I just tried to make things very general so that they could fit into diverse clients' needs. But I'm not so sure about the Uber/Lyft part and the scheduling algorithm. We would have to think that through.